File: src/main/java/com/jss/osiris/modules/osiris/crm/service/CandidacyService.java
package com.jss.osiris.modules.osiris.crm.service;

import java.util.List;

import com.jss.osiris.libs.exception.OsirisException;
import com.jss.osiris.modules.osiris.crm.model.Candidacy;

public interface CandidacyService {
    public List<Candidacy> getCandidacies();

    public List<Candidacy> getCandidacies(Boolean isDisplayTreated);

    public Candidacy getCandidacy(Integer id);

    public Candidacy declareNewCandidacy(Candidacy candidacy) throws OsirisException;

    public Candidacy addOrUpdateCandidacy(Candidacy candidacy) throws OsirisException;

    public void markCandidacyAsUnTreated(Candidacy candidacy) throws OsirisException;

    public void markCandidacyAsTreated(Candidacy candidacy) throws OsirisException;
}

File: client/src/app/modules/crm/services/candidacy.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Candidacy } from '../../crm/model/Candidacy';

@Injectable({
  providedIn: 'root'
})
export class CandidacyService extends AppRestService<Candidacy> {

  constructor(http: HttpClient) {
    super(http, "crm");
  }

  getCandidacies(isDisplayTreated: boolean) {
    return this.getList(new HttpParams().set("isDisplayTreated", isDisplayTreated), "candidacies");
  }

  markCandidacyAsTreated(candidacy: Candidacy) {
    return this.get(new HttpParams().set("idCandidacy", candidacy.id), "candidacy/treated")
  }

  markCandidacyAsUnTreated(candidacy: Candidacy) {
    return this.get(new HttpParams().set("idCandidacy", candidacy.id), "candidacy/untreated")
  }

}
